### PR TITLE
Refactor: Simplify Profile Metadata System

### PR DESCRIPTION
**Pull request by Zer0dot**
This PR simplifies the profile metadata system to be only on a per-profile basis, removing the per-user abstraction layer.

### CODE REVIEW

File: contracts/misc/LensPeriphery.sol [hunk 1]
```diff
@@ -36,59 +36,60 @@ contract LensPeriphery {
...
      * @param profileId The profile ID to set the metadata for.
-     * @param metadata The metadata string to set for the profile and message sender.
+     * @param metadata The metadata string to set for the profile.
      */
     function setProfileMetadataURI(uint256 profileId, string calldata metadata) external {
-        _setProfileMetadataURI(msg.sender, profileId, metadata);
+        address owner = IERC721Time(address(HUB)).ownerOf(profileId);
+        if (msg.sender != owner) revert Errors.NotProfileOwner();
+        _setProfileMetadataURI(profileId, metadata);
     }
```
Comment (donosonaumczuk) [inline contracts/misc/LensPeriphery.sol:66]: Maybe for consistency with LensHub makes sense to unify them into a single `setProfileMetadataURI` function that does the validation ensuring the sender is the owner or its dispatcher. Only disadvantage I see is a bit more of gas.

**Review by donosonaumczuk — CHANGES_REQUESTED** [review]
Small doubt/comment but looks good

File: contracts/misc/LensPeriphery.sol (hunk 1 above)
Comment (Zer0dot) [reply contracts/misc/LensPeriphery.sol]: Done in https://github.com/aave/lens-protocol/pull/112/commits/0d72b2fffcee19783de6e86e5c5a407bf388d208!

**Review by donosonaumczuk — APPROVED** [review]
Looks good, approving

**Pull request merged**